torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -522,14 +522,38 @@ Tensor clamp_backward(const Tensor & grad, const Tensor &self, const optional<Sc
   }
 }
 
-Tensor mm_mat1_backward(const Tensor & grad, const Tensor & mat2, const Tensor & mat1, const Scalar & alpha) {
-  // if input was column-major, return grad as column-order for efficiency
-  if (mat1.is_sparse()) {
-    throw std::runtime_error("calculating the gradient of a sparse Tensor argument to mm is not supported.");
+// This function is used by load_derivatives.py to replace tensor.strides()
+// calls that appear in derivative formulas. If the tensor has requires_grad
+// set, this function returns its strides or throws an error if the tensor
+// is sparse. If requires_grad is not set, an empty array is returned since
```

Comment:
Note that, as I discovered today, this is not true for all inplace ops:
```
a = torch.rand(4)
a.copy_(torch.rand(4, requires_grad=True))
```
In the `copy_` op, the input does not require grad but a backward can actually happen.

So the right thing to check here would be: `differentiable functions && (arg.requires_grad || is_modified_inplace(arg))`.
But here again I think adding an assert will be good here. Even though it is going to be tricky as you don't know yet in `load_derivative` if the function is inplace or not.
The only place is [here](https://github.com/pytorch/pytorch/blob/9f0dbc1e07593cb001d2d15f394383a48737799d/tools/autograd/load_derivatives.py#L404) where you can check `declaration['inplace']`. But then you need to mark the formula as not usable for inplace ops. And add a check there to make sure the formula we get is usable for inplace when the declaration is inplace.

Reply:
I added an assert to chech `declaration['inplace']`, and then throw an error if the `self` derivative formula has the `strides_or_error` replacement. This should only apply for the `self` derivative, right?

Reply:
Yes, all the differentiable inplace ops change the "self" argument AFAIK